ticdc: add doc for row filter config (#9759)
pingcap/docs: ticdc/manage-ticdc.md

protocol = "canal-json"
```

### Event filter rules <span class="version-mark">New in v6.2.0</span>

Starting in v6.2.0, TiCDC supports event filter. You can configure event filter rules to filter out the DML and DDL events that meet the specified conditions.

The following is an example of event filter rules:

```toml
[filter]

> # The event filter rules must be under the filter configuration. You can configure multiple event filters at the same time.

Reviewer: ```suggestion
# The event filter rules must be under the `[filter]` configuration. You can configure multiple event filters at the same time.
```

[[filter.event-filters]]
matcher = ["test.worker"] # matcher is an allow list, which means this rule only applies to the worker table in the test database.
ignore-event = ["insert"] # Ignore insert events.

> ignore-sql = ["^drop", "add column"] # Ignore DDLs that start with "drop" or contain "add column".

Reviewer: 这里的逗号是 “或” 的关系还是 “和” 的关系？可能需要开发确认下。

Author: @asddongmen 麻烦看下？

Reviewer 2: `or`
例如 "drop table t1" 会被忽略， "alter table t1 add column age (int)" 也会被忽略，这里的 `or` 的意思是满足条件之一就会忽略，我认为这个描述是没问题的。

ignore-delete-value-expr = "name = 'john'" # Ignore delete DMLs that contain the condition "name = 'john'".
ignore-insert-value-expr = "id >= 100" # Ignore insert DMLs that contain the condition "id >= 100".
ignore-update-old-value-expr = "age < 18 or name = 'lili'" # Ignore update DMLs whose old value contains "age < 18" or "name = 'lili'".
ignore-update-new-value-expr = "gender = 'male' and age > 18" # Ignore update DMLs whose new value contains "gender = 'male'" and "age > 18".
```

The event filter rules must be under the `[filter]` configuration. For detailed configuration, refer to [Task configuration file](#task-configuration-file).

> Configuration parameters description:

Reviewer: ```suggestion
Description of configuration parameters :
```

- `matcher`: the database and table that this event filter rule applies to. The syntax is the same as [table filter](/table-filter.md).
- `ignore-event`: the event type to be ignored. This parameter accepts an array of strings. You can configure multiple event types. Currently, the following event types are supported:

| Event           | Type | Alias | Description         |
| --------------- | ---- | -|--------------------------|
| all dml         |      | |Matches all DML events       |
| all ddl         |      | |Matches all DDL events         |
| insert          | DML  | |Matches `insert` DML event      |
| update          | DML  | |Matches `update` DML event      |
| delete          | DML  | |Matches `delete` DML event      |
| create schema   | DDL  | create database |Matches `create database` event |
| drop schema     | DDL  | drop database  |Matches `drop database` event |
| create table    | DDL  | |Matches `create table` event    |
| drop table      | DDL  | |Matches `drop table` event      |
| rename table    | DDL  | |Matches `rename table` event    |
| truncate table  | DDL  | |Matches `truncate table` event  |

> | alter table     | DDL  | |Matches `alter table` event, including all clauses of `alter table` and `create index`/`drop index`   |

Reviewer: ```suggestion
| alter table     | DDL  | |Matches `alter table` event, including all clauses of `alter table`, `create index` and `drop index`   |
```

| add table partition    | DDL  | |Matches `add table partition` event     |
| drop table partition    | DDL  | |Matches `drop table partition` event     |
| truncate table partition    | DDL  | |Matches `truncate table partition` event     |
| create view     | DDL  | |Matches `create view`event     |
| drop view     | DDL  | |Matches `drop view` event     |

- `ignore-sql`: the DDL statements to be ignored. This parameter accepts an array of strings, in which you can configure multiple regular expressions. This rule only applies to DDL events.
- `ignore-delete-value-expr`: this parameter accepts a SQL expression. This rule only applies to delete DML events with the specified value.
- `ignore-insert-value-expr`: this parameter accepts a SQL expression. This rule only applies to insert DML events with the specified value.
- `ignore-update-old-value-expr`: this parameter accepts a SQL expression. This rule only applies to update DML events whose old value contains the specified value.
- `ignore-update-new-value-expr`: this parameter accepts a SQL expression. This rule only applies to update DML events whose new value contains the specified value.

> **Note:**
>

> > - When TiDB updates a value in the column of the clustered index, TiDB splits a UPDATE event into a DELETE event and an INSERT event. TiCDC does not identify such events as an UPDATE event and thus cannot correctly filter out such events.

Reviewer: ```suggestion
> - When TiDB updates a value in the column of the clustered index, TiDB splits an `UPDATE` event into a `DELETE` event and an `INSERT` event. TiCDC does not identify such events as an `UPDATE` event and thus cannot correctly filter out such events.
```

> > - When you configure a SQL expression, make sure all tables that matches `matcher` contain the columns specified in the SQL expression. Otherwise, the replication task cannot be created. In addition, if the table schema changes during the replication, which results in a table no longer containing a required column, the replication task enters an error state that cannot be resumed automatically. In such a situation, you must manually modify the configuration and resume the task.

Reviewer: ```suggestion
> - When you configure a SQL expression, make sure all tables that matches `matcher` contain all the columns specified in the SQL expression. Otherwise, the replication task cannot be created. In addition, if the table schema changes during the replication, which results in a table no longer containing a required column, the replication task fails and cannot be resumed automatically. In such a situation, you must manually modify the configuration and resume the task.
```

### Notes for compatibility

* In TiCDC v4.0.0, `ignore-txn-commit-ts` is removed and `ignore-txn-start-ts` is added, which uses start_ts to filter transactions.
* In TiCDC v4.0.2, `db-dbs`/`db-tables`/`ignore-dbs`/`ignore-tables` are removed and `rules` is added, which uses new filter rules for databases and tables. For detailed filter syntax, see [Table Filter](/table-filter.md).

> * In TiCDC v6.1.0, `mounter` is removed. If you configure `mounter`, TiCDC does not report an error, but the configuration does not take effect.

Reviewer: ```suggestion
* In TiCDC v6.1.0 and later versions, `mounter` is removed. If you configure `mounter`, TiCDC does not report an error, but the configuration does not take effect.
```
